platform/mellanox/docker-syncd-mlnx/ecmp_calc_sdk.py

```diff
@@ -0,0 +1,466 @@
+        if rc == SX_STATUS_ENTRY_NOT_FOUND:
+            return 0
+        sx_check_rc(rc)
+        
+        entry = sx_uc_route_get_entry_t_arr_getitem(entries_array, 0)
+        if entry.route_data.type == SX_UC_ROUTE_TYPE_NEXT_HOP:
+            return entry.route_data.uc_route_param.ecmp_id
+        
+        return 0
+    
+    finally:
+        delete_sx_ip_prefix_t_p(ip_prefix_p)
+        delete_uint32_t_p(entries_cnt_p)
+        delete_sx_uc_route_get_entry_t_arr(entries_array)
+
+def sx_router_operational_ecmp_get(handle, ecmp_id):
+    """ Get next hops for a given ECMP id
+    
+    Args:
+        handle (sx_api_handle_t): SDK handle
+        ecmp_id (int): ECMP id
+    
+    Returns:
+        list: List of next hops
+    """    
+    try:
+        next_hops = []
+        
+        next_hop_count_p = new_uint32_t_p()
+        uint32_t_p_assign(next_hop_count_p, 0)
+        
+        rc = sx_api_router_operational_ecmp_get(handle, ecmp_id, None, next_hop_count_p)
+        sx_check_rc(rc)
+        
+        next_hop_count  = uint32_t_p_value(next_hop_count_p)
+        next_hop_list_p = new_sx_next_hop_t_arr(next_hop_count)
+        
+        rc = sx_api_router_operational_ecmp_get(handle, ecmp_id, next_hop_list_p, next_hop_count_p)
+        sx_check_rc(rc)
+        
+        next_hop_count  = uint32_t_p_value(next_hop_count_p)
+        for i in range(next_hop_count):
+            next_hop = sx_next_hop_t_arr_getitem(next_hop_list_p, i)
+            if next_hop.next_hop_key.type == SX_NEXT_HOP_TYPE_IP:
+                next_hops.append(next_hop)
+                
+        return next_hops
+    
+    finally:
+        delete_uint32_t_p(next_hop_count_p)
+        delete_sx_next_hop_t_arr(next_hop_list_p)
+
+def sx_get_router_interface(handle, vrid, rif):
+    """ Get router interface information
+    
+    Args:
+        handle (sx_api_handle_t): SDK handle
+        vrid (sx_router_id_t): virtual router id
+        rif (sx_router_interface_t): router interface id
+    
+    Returns:
+        dict : Dictionary contains interface parameters
+    """       
+    try:
+        vrid_p = new_sx_router_id_t_p()
+        sx_router_id_t_p_assign(vrid_p, vrid)
+    
+        ifc_p = new_sx_router_interface_param_t_p()
+        ifc_attr_p = new_sx_interface_attributes_t_p()
+        rif_params = {}
+        
+        rc = sx_api_router_interface_get(handle, rif, vrid_p, ifc_p, ifc_attr_p)
+        sx_check_rc(rc)
+        
+        if ifc_p.type == SX_L2_INTERFACE_TYPE_PORT_VLAN:
+            rif_params['port'] = ifc_p.ifc.port_vlan.port
+            
+        if ifc_p.type == SX_L2_INTERFACE_TYPE_VPORT:
+            rif_params['vport'] = ifc_p.ifc.vport.vport        
+    
+        if ifc_p.type == SX_L2_INTERFACE_TYPE_VLAN:
+            rif_params['vlan'] = ifc_p.ifc.vlan.vlan
+            
+        return rif_params
+        
+    finally:
+        delete_sx_router_id_t_p(vrid_p)
```

Comment:
@liorghub free is opposite to construction

Reply:
Done.